docs: add nocloud documentation (#4377)
siderolabs/talos: website/content/docs/v0.13/Cloud Platforms/nocloud.md

description: "Creating a cluster via the CLI using qemu."
---

Talos supports [nocloud](https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html) data source implementation.
There are two ways to configure your server:

* SMBIOS “serial number” option
* CDROM or USB-flash filesystem

### SMBIOS

This method works only with network.
DHCP and HTTP(s) server required.

> ```

Author: Yep, I've add the screenshot bellow

description: "Creating a cluster via the CLI using qemu."
---

Talos supports [nocloud](https://cloudinit.readthedocs.io/en/latest/topics/datasources/nocloud.html) data source implementation.
There are two ways to configure your server:

* SMBIOS “serial number” option
* CDROM or USB-flash filesystem

### SMBIOS

This method works only with network.
DHCP and HTTP(s) server required.

```
ds=nocloud-net;s=http://10.10.0.1/configs/;h=HOSTNAME
```

After the network initialization is complete, Talos fetches:

* the machine config from http://10.10.0.1/configs/user-data
* the network config (if it exists) from http://10.10.0.1/configs/network-config

#### QEMU

Simply add the following flag when starting the VM.

```
-smbios type=1,serial=ds=nocloud-net;s=http://10.10.0.1/configs/
```

#### Proxmox

Set the source machine config throught the serial number on Proxmox GUI.

> <img src="/images/proxmox-guide/proxmox-smbios.png" width="920px">

Reviewer: wonder if it is easier to identify if it is in a new no-cloud directory

> The Proxmox stores the VM config here /etc/pve/qemu-server/$ID.conf (```$ID``` - VM ID number of virtual machine), you will see someting like:

Reviewer: ```suggestion
The Proxmox stores the VM config at /etc/pve/qemu-server/$ID.conf (```$ID``` - VM ID number of virtual machine), you will see something like:
```

```conf
...
smbios1: uuid=ceae4d10,serial=ZHM9bm9jbG91ZC1uZXQ7cz1odHRwOi8vMTAuMTAuMC4xL2NvbmZpZ3Mv,base64=1
...
```

Where serial is a base64-encoded string of `ds=nocloud-net;s=http://10.10.0.1/configs/`.

### CDROM/USB

> Talos can also get machine config from local storage without running a network services (DHCP/HTTP).

Reviewer: ```suggestion
Talos can also get machine config from local storage without running network services (DHCP/HTTP).
```

You can provide configs to the server via files on a vfat or iso9660 filesystem. The filesystem volume label must be ```cidata``` or ```CIDATA```.

#### QEMU

Create and prepare Talos machine config:

```bash
export CONTROL_PLANE_IP=192.168.1.10

talosctl gen config talos-nocloud https://$CONTROL_PLANE_IP:6443 --output-dir _out
```

Prepare cloud-init configs:

```bash
mkdir -p iso
mv _out/controlplane.yaml iso/user-data
echo "local-hostname: controlplane-1" > iso/meta-data
cat > iso/network-config << EOF
version: 1
config:
   - type: physical
     name: eth0
     mac_address: "52:54:00:12:34:00"
     subnets:
        - type: static
          address: 192.168.1.10
          netmask: 255.255.255.0
          gateway: 192.168.1.254
EOF
```

Create cloud-init iso image

```bash 
cd iso && genisoimage -output cidata.iso -V cidata -r -J user-data meta-data network-config
```

Start the VM

```
qemu-system-x86_64 \
    ...
    -cdrom iso/cidata.iso \
    ...
```

#### Proxmox

Proxmox can create cloud-init disk for you.
[Original documentation](https://pve.proxmox.com/wiki/Cloud-Init_Support).

> Edit the config file as follows, substituting your own infomation as necessary:

Reviewer: ```suggestion
Edit the cloud-init config information in proxmox as follows,  substitute your own information as necessary:
```
